ebos/eclwriter.hh

```diff
@@ -743,14 +695,80 @@ private:
     const Opm::Schedule& schedule() const
     { return simulator_.vanguard().schedule(); }
 
+    void prepareLocalCellData(const bool isSubStep,
+                              const int  reportStepNum)
+    {
+        const auto& gridView = simulator_.vanguard().gridView();
+        const int numElements = gridView.size(/*codim=*/0);
+        const bool log = collectToIORank_.isIORank();
+
+        eclOutputModule_.allocBuffers(numElements, reportStepNum,
+                                      isSubStep, log, /*isRestart*/ false);
+
+        ElementContext elemCtx(simulator_);
+        ElementIterator elemIt = gridView.template begin</*codim=*/0>();
+
+        const ElementIterator& elemEndIt = gridView.template end</*codim=*/0>();
+        for (; elemIt != elemEndIt; ++elemIt) {
+            const Element& elem = *elemIt;
+
+            elemCtx.updatePrimaryStencil(elem);
+            elemCtx.updatePrimaryIntensiveQuantities(/*timeIdx=*/0);
+
+            eclOutputModule_.processElement(elemCtx);
+        }
+    }
+
+    void writeOutput(const int               reportStepNum,
+                     const bool              isSubStep,
+                     ::Opm::data::Solution&& localCellData,
+                     ::Opm::data::Wells&&    localWellData)
+    {
+        const Scalar curTime = simulator_.time() + simulator_.timeStepSize();
+        const Scalar nextStepSize = simulator_.problem().nextTimeStepSize();
+        const auto isParallel = this->collectToIORank_.isParallel();
+
+        Opm::RestartValue restartValue {
+            isParallel ? this->collectToIORank_.globalCellData()
+                       : std::move(localCellData),
+
+            isParallel ? this->collectToIORank_.globalWellData()
+                       : std::move(localWellData)
+        };
+
+        if (simulator_.vanguard().eclState().getSimulationConfig().useThresholdPressure()) {
+            restartValue.addExtra("THRESHPR", Opm::UnitSystem::measure::pressure,
+                                  simulator_.problem().thresholdPressure().data());
+        }
+
+        // Add suggested next timestep to extra data.
+        if (! isSubStep) {
+            restartValue.addExtra("OPMEXTRA", std::vector<double>(1, nextStepSize));
+        }
+
+        // first, create a tasklet to write the data for the current time
+        // step to disk
+        auto eclWriteTasklet = std::make_shared<EclWriteTasklet>(
+            this->actionState(), this->summaryState(), *this->eclIO_,
```

Comment:
Please reformat with one argument per line.

Reply:
> Please reformat with one argument per line.

I'll do that the next time I'm touching this file.